EXAMPLES/src/clientUsage/ClientMethodChaining.re

```diff
@@ -0,0 +1,79 @@
+module ApolloQueryResult = ApolloClient.Types.ApolloQueryResult;
+module FetchResult = ApolloClient.Types.FetchResult;
+
+module AddTodoMutation = [%graphql
+  {|
+    mutation AddTodo($text: String!) {
+      todo: addTodoSimple(text: $text) {
+        id
+        text
+      }
+    }
+  |}
+];
+
+module TodosQuery = [%graphql
+  {|
+    query TodosQuery {
+      todos: allTodos {
+        id
+        text
+        completed
+      }
+    }
+  |}
+];
+let fauxTodosQuery = ();
+let fauxAddTodoMutation = ();
+
+type fauxVariables = {text: string};
+type fauxJsClient('query, 'mutation) = {
+  query:
+    (~query: 'query, unit) =>
+    Js.Promise.t(ApolloClient.Types.ApolloQueryResult.t(Js.Json.t)),
+  mutate:
+    (~mutation: 'mutation, fauxVariables) =>
+    Js.Promise.t(ApolloClient.Types.FetchResult.t(Js.Json.t)),
+};
+
+let client = Client.instance;
+[@bs.val] external jsClient: fauxJsClient('query, 'mutation) = "jsClient";
+
+client.query(~query=(module TodosQuery), ())
+->Promise.flatMapOk(({data}) => {
+    client.mutate(~mutation=(module AddTodoMutation), {text: ""})
+  })
+->Promise.get(result =>
+    switch (result) {
+    | Ok(_) => Js.log("Success for both!")
+    | Error(error) =>
+      Js.log2("Typed ApolloError if one of them failed!", error)
+    }
+  );
```

Comment:
This is an ideal situation, imo. It's clean and you're never going to do one of these `query |> Js.Promise.then_(_ => Js.log("We're good!"))`.

Reply:
One downside of this is you now need to also go learn `reason-promise`, but I find most new people copy/paste and they can probably get pretty far doing that when things are this simple?